### PR TITLE
feat: types for `react-devtools-inline`

### PR DESCRIPTION
**Pull request by YardenPorat**
This PR adds typings for `react-devtools-inline`.
`react-devtools-inline` imports and uses types which are defined at `react-devtools-shared` and  `react-devtools-timeline`, which are private packages (as seen [here](https://github.com/facebook/react/blob/cf665c4b73a28b034c8173f4d929205fb12d2d2e/packages/react-devtools-shared/package.json#L2) and [here](https://github.com/facebook/react/blob/cf665c4b73a28b034c8173f4d929205fb12d2d2e/packages/react-devtools-timeline/package.json#L2)).
The choice I have made regarding type within this private package is to add those who are in use to a `commons.d.ts` file

----------------

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
was this autogenerated, so those 'undefined' types in params could not be avoided?

**Comment by AviVahl** [comment]
@peterblazejewicz  These were manually generated (@YardenPorat and I are working together). The explicit `undefined` was added to better support projects with `exactOptionalPropertyTypes` turned on. Similar to the PRs by @sandersn: https://github.com/DefinitelyTyped/DefinitelyTyped/pulls?q=is%3Apr+author%3Asandersn+is%3Aclosed+exactOptionalPropertyTypes back in the day.

**Comment by peterblazejewicz** [comment]
Those are not required for input params, those are for props (so on class/interface/type/literals)> I understand intent, which is OK for non-input.

**Comment by AviVahl** [comment]
I see, so specifically for function/method parameters. Makes sense. We'll get the types updated and ping you back. Thanks for reviewing!

**Comment by YardenPorat** [comment]
Thanks for reviewing
Removed `undefined` for optional arguments of function\methods.
Also removed `"redundant-undefined": false,` from `tslint.json` file

File: types/react-devtools-inline/commons.d.ts [hunk 1]
```diff
@@ -0,0 +1,1303 @@
...
+        didError?: boolean,
+    ) => void;
+    /** Timeline internal module filtering */
+    getInternalModuleRanges: () => Array<[string, string]>;
+    registerInternalModuleStart: (moduleStartError: Error) => void;
+    registerInternalModuleStop: (moduleStopError: Error) => void;
+    dangerous_setTargetConsoleForTesting?: (fakeConsole: Record<string, unknown>) => void | undefined;
+}
+
+declare var __REACT_DEVTOOLS_GLOBAL_HOOK__: DevToolsHook;
```
Comment (peterblazejewicz) [inline types/react-devtools-inline/commons.d.ts:1298]:
```suggestion
```
do you need this one? It's not intended to be used via import, just as global via window object (per documentation). That would clean things out, including exclusion for `tslint`?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by YardenPorat** [comment]
Ready to merge

**Comment by YardenPorat** [comment]
> LGTM!

Thanks Piotr for your quick response and reviewal 😃

**Pull request merged**